spec/requests/bulk_upload_lettings_logs_controller_spec.rb

```diff
@@ -73,5 +73,116 @@
         expect(response.body).to include("How to upload logs in bulk")
       end
     end
+
+    context "when no year is specified" do
+      it "shows guidance page with links defaulting to the current year" do
+        get "/lettings-logs/bulk-upload-logs/guidance"
+
+        expect(response.body).to include("Download the lettings bulk upload template (#{current_collection_start_year} to #{current_collection_start_year + 1})")
+      end
+    end
+
+    context "when an invalid year is specified" do
+      it "shows not found" do
+        get "/lettings-logs/bulk-upload-logs/guidance?form%5Byear%5D=10000"
+
+        expect(response).to be_not_found
+      end
+    end
+  end
+
+  describe "GET /lettings-logs/bulk-upload-logs/year" do
+    it "does not require a year to be specified" do
+      get "/lettings-logs/bulk-upload-logs/year"
+
+      expect(response).to be_ok
+    end
+  end
+
+  pages_requiring_year_specification = %w[prepare-your-file upload-your-file checking-file]
+  pages_requiring_year_specification.each do |page_id|
+    describe "GET /lettings-logs/bulk-upload-logs/#{page_id}" do
+      context "when no year is provided" do
+        it "returns not found" do
+          get "/lettings-logs/bulk-upload-logs/#{page_id}"
+
+          expect(response).to be_not_found
+        end
+      end
+
+      context "when requesting the previous year in a crossover period" do
+        before do
+          allow(FormHandler.instance).to receive(:lettings_in_crossover_period?).and_return(true)
+        end
+
+        it "succeeds" do
+          get "/lettings-logs/bulk-upload-logs/#{page_id}?form%5Byear%5D=#{current_collection_start_year - 1}"
+
+          expect(response).to be_ok
+        end
+      end
+
+      context "when requesting the previous year outside a crossover period" do
+        before do
+          allow(FormHandler.instance).to receive(:lettings_in_crossover_period?).and_return(false)
+        end
+
+        it "returns not found" do
+          get "/lettings-logs/bulk-upload-logs/#{page_id}?form%5Byear%5D=#{current_collection_start_year - 1}"
+
+          expect(response).to be_not_found
+        end
+      end
+
+      context "when requesting the current year" do
+        it "succeeds" do
+          get "/lettings-logs/bulk-upload-logs/#{page_id}?form%5Byear%5D=#{current_collection_start_year}"
+
+          expect(response).to be_ok
+        end
+      end
+
+      if page_id != "prepare-your-file"
```

Comment:
The prepare your file page has a year based switch statement for the view rather than just passing in the year, so we can expect this test to fail for that page whenever the next year hasn't been set up yet - which means we really don't want this test in that case.